site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+
+            // Get name of the command which we are looking for
+            const targetCmdName = interaction.data.options?.[0].options?.[0].name ||
+              interaction.data.options?.[0].options?.[0].name;
+            if (!targetCmdName) return;
+
+            // Try to find the command
+            command = subCommandGroup.subCommands.find((c) => c.name === targetCmdName);
+
+            commandName += ` ${subCommandGroup.name} ${command?.name}`;
+
+            // Normal
+          }
+
+          if (optionType === ApplicationCommandOptionTypes.SubCommandGroup) {
+            // Check if command has subcommand and handle types
+            if (!command?.subcommands) return;
+
+            // Try to find the command
+            const found = command.subcommands.find((command) => command.name == interaction.data?.options?.[0].name);
+            if (!found) return;
+
+            if (isSubCommandGroup(found)) return;
+
+            command = found;
+            commandName += ` ${command?.name}`;
+          }
+        }
+
+        // Due to the nature of commands, we want to make sure that one failed command won't crash the whole bot.
+        try {
+          if (command) {
+            await command.execute(rawBot, interaction);
+            log.info(
+              `[Command: ${bgYellow(black(String(interaction.data.name)))} - ${
+                bgBlack(green(`Success`))
+              }] by ${interaction.user.username}#${interaction.user.discriminator} in ${guildName}${
+                guildName !== "Direct Message" ? ` (${guild.id})` : ``
+              }`,
+            );
+          } else {
+            throw "";
+          }
+          // Speaking of failed commands, this will just print out the error and move on:
+        } catch (err) {
+          log.error(
+            `[Command: ${bgYellow(black(String(interaction.data.name)))} - ${
+              bgBlack(red(`Error`))
+            }] by ${interaction.user.username}#${interaction.user.discriminator} in ${guildName}${
+              guildName !== "Direct Message" ? ` (${guild.id})` : ``
+            }`,
+          );
+          err.length ? log.error(err) : undefined;
+        }
+        // Other than that, we should log that we couldn't find the command:
+      } else {
+        log.warn(
+          `[Command: ${bgYellow(black(String(interaction.data.name)))} - ${
+            bgBlack(yellow(`Not Found`))
+          }] by ${interaction.user.username}#${interaction.user.discriminator} in ${guildName}${
+            guildName !== "Direct Message" ? ` (${guild.id})` : ``
+          }`,
+        );
+      }
+    }
+  }
+};
+```
+
+That might seem scary, however, we've included a lot of comments to help you understand what it does. We also need to
+add some more helpers to the `src/utils/helpers.ts` file (make sure to import types from `src/commands/mod.ts` and
+`deps.ts`):
+
+```typescript title="src/utils/helpers.ts"
+// ...
+
+export async function getGuildFromId(
```

Comment:
Why is this needed?

Reply:
As in the `// ...`? That would signify earlier file contents. As in why `getGuildFromId()` is needed? I remember that the bot's helper for this won't automatically add the guild to the cache, therefore we use this function instead.